docker-proxy-rule-core-jdk21/build.gradle

```diff
@@ -1,12 +1,24 @@
 apply plugin: 'com.palantir.external-publish-jar'
+apply plugin: 'org.unbroken-dome.test-sets'
+
+testSets {
+    integrationTest
+}
+
+build.dependsOn integrationTest
 
 dependencies {
     api project(':docker-proxy-rule-core')
 
-    testImplementation group: 'junit', name: 'junit'
-    testImplementation group: 'org.assertj', name: 'assertj-core'
-    testImplementation group: 'org.mockito', name: 'mockito-core'
-    testRuntimeOnly group: 'org.mockito', name: 'mockito-inline'
+    compileOnly group: 'com.google.auto.service', name: 'auto-service'
```

Comment:
we can use `name: 'auto-service-annotations'` here to reduce the scope of what's on the compilation classpath -- no need to bring along the annotation processor itself, only `AutoService.class` :-)